File: 884. Uncommon Words from Two Sentences/solution.py
class Solution:
    def uncommonFromSentences(self, A: str, B: str) -> List[str]:
        a_list = A.split(" ")
        b_list = B.split(" ")
        a_dict = dict()
        b_dict = dict()
        for a in a_list:
            if a not in a_dict:
                a_dict[a] = 1
            else:
                a_dict[a] += 1

        for b in b_list:
            if b not in b_dict:
                b_dict[b] = 1
            else:
                b_dict[b] += 1

        res = []
        for a in a_list:
            if a not in b_dict and a_dict[a] == 1:
                res.append(a)

        for b in b_list:
            if b not in a_dict and b_dict[b] == 1:
                res.append(b)
        return res

File: 884. Uncommon Words from Two Sentences/solution.go
import (
	"strings"
)

func uncommonFromSentences(A string, B string) []string {
	aList := strings.Split(A, " ")
	bList := strings.Split(B, " ")
	aDict := make(map[string]int)
	bDict := make(map[string]int)
	for _, a := range aList {
		if _, ok := aDict[a]; ok {
			aDict[a]++
		} else {
			aDict[a] = 1
		}
	}
	for _, b := range bList {
		if _, ok := bDict[b]; ok {
			bDict[b]++
		} else {
			bDict[b] = 1
		}
	}
	res := make([]string, 0)
	for _, a := range aList {
		if _, ok := bDict[a];! ok && aDict[a] == 1 {
			res = append(res, a)
		}
	}

	for _, b := range bList {
		if _, ok := aDict[b]; ! ok && bDict[b] == 1 {
			res = append(res, b)
		}
	}
	return res
}
